### PR TITLE
[SYCL][XPTI] SYCL use of XPTI - documentation update

### PR DESCRIPTION
**Pull request by tovinkere**
This PR documents the trace events generated by SYCL runtime and the description of the callback parameters. Subscribers implementing tools that rely on this tracing data are primary users.

### CODE REVIEW

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
...: `nullptr`</li> <li> **event**: The event ID will reflect the ID of the command group object submission that created this event or a new event based on the combination of the string "queue.wait" and the address of the event.  </li> <li> **instance**: Unique ID to allow the correlation of the `wait_begin` event with the `wait_end` event. </li> <li> **user_data**: String indicating `event.wait` and the address of the event sent in as `const char *` </li> <p></p>Tracing the `queue.wait()` or `queue.wait_and_throw()` will capture the waiting on the action represented by the event object, which could be the execution of a kernel, completion of a memory operation, etc that is embedded in the command group handler. All wait events contain metadata that indicates the SYCL device on which the corresponding operation has been submitted. If the event is from a command group handler, then the source location information is available as well.    | **`sycl_device`**, `sym_function_name`, `sym_source_file_name`, `sym_line_no`                                                                                                                                                                            |
```
Comment (alexbatashev) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md:243]: Are these supposed to be `barrier_begin` and `barrier_end`?

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+>the [Framework Documentation](https://github.com/intel/llvm/tree/sycl/xptifw/doc/XPTI_Framework.md) for API use
+>and framework performance data.
+
+This document outlines the use of this framework API in the SYCL runtime
+library. The primary concept enable by this framework is the generation of a
+unique 64-bit ID, referred to as the Universal ID (UID), for every public
+language entry point into the library. This allows tools and other helps in the
+software stack to correlate debug and performance data by tagging it with the
+64-bit UID. The framework also provides the ability to propagate this UID all
+the way to the driver layers for the target device so data from lower layers and hardware can be correlated easily.
```
Comment (alexbatashev) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md:18]: Could you please shrink this line to 80 characters?

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+The XPTI concepts in use here are:
+
+1.  Tracepoints - define all the points in a software layer we want to instrument or trace. The trace point is used to generate the UID.
```
Comment (alexbatashev) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md:22]: And these three bullets, too?

**Review by alexbatashev — COMMENTED** [review]
Thanks a lot for sharing this!

File: sycl/doc/SYCLInstrumentationUsingXPTI.md (hunk 1 above)
Comment (tovinkere) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md:243]: Yes, wait_begin/wait_end are used to model barriers. In SYCL, these events are generated from event.wait() and similar calls.

File: sycl/doc/SYCLInstrumentationUsingXPTI.md (hunk 2 above)
Comment (tovinkere) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md]: Done!

File: sycl/doc/SYCLInstrumentationUsingXPTI.md (hunk 3 above)
Comment (tovinkere) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md]: Done!

File: sycl/doc/SYCLInstrumentationUsingXPTI.md (hunk 1 above)
Comment (alexbatashev) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md:247]: @tovinkere I mean wait_begin and wait_end are repeated twice, but barriers, that are also [reported](https://github.com/intel/llvm/blob/7735139bb4d8420fcde692b6eddd2ebf9257efe8/sycl/source/detail/scheduler/commands.cpp#L607) are not mentioned.

Comment (tovinkere) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md]: Ah, thanks - good catch! barrier_begin/barrier_end trace events were not documented. wait_begin/wait_end has two flavors and they have been documented, but I had to make some changes to add clarity.

**Review by alexbatashev — APPROVED** [review]

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
+# SYCL Instrumentation
```
Comment (bader) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md:1]:
```suggestion
# SYCL Instrumentation

```

**Review by bader — DISMISSED** [review]

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@tovinkere, thanks a lot for your contribution! I just wanted to remind that it's preferable to open a pull request using a private fork branch to keep the repository clean from private branches with unfinished work.

**Pull request merged**